agent/app/agent.go

```diff
@@ -298,17 +298,36 @@ func (agent *ecsAgent) doStart(containerChangeEventStream *eventstream.EventStre
 			return exitcodes.ExitTerminal
 		}
 	}
+	numGPUs := int64(0)
 	if agent.cfg.GPUSupportEnabled {
 		err := agent.initializeGPUManager()
 		if err != nil {
 			seelog.Criticalf("Could not initialize Nvidia GPU Manager: %v", err)
 			return exitcodes.ExitError
 		}
+		// Find number of GPUs instance has
+		platformDevices := agent.getPlatformDevices()
+		for _, device := range platformDevices {
+			if *device.Type == ecs.PlatformDeviceTypeGpu {
+				numGPUs++
+			}
+		}
+	}
+
+	hostResources, err := client.GetHostResources()
+	if err != nil {
+		seelog.Critical("Unable to fetch host resources")
+		return exitcodes.ExitError
```

Comment:
Discussed with @prateekchaudhry offline. We should revisit this error code by referring logic [here](https://github.com/aws/amazon-ecs-agent/blob/883211e24d9255023a68f6351001c291af8a699b/agent/app/agent.go#L391-L398).
* [ExitError](https://github.com/aws/amazon-ecs-agent/blob/883211e24d9255023a68f6351001c291af8a699b/agent/sighandlers/exitcodes/exitcodes.go#L21-L23): (as well as unspecified exit codes) indicates a fatal error occurred, but the agent should be restarted
* [ExitTerminal](https://github.com/aws/amazon-ecs-agent/blob/883211e24d9255023a68f6351001c291af8a699b/agent/sighandlers/exitcodes/exitcodes.go#LL24C2-L26C18): indicates the agent has exited unsuccessfully, but should not be restarted

Reply:
Discussed with @yinyic , we probably should use ExitError because it may be fail to read the host resource due to some temporary reasons and a retry will work.